src/compiler/interface.jl

```diff
@@ -11,6 +11,27 @@ Context() = Context(nothing)
 
 cache(cx::Context) = cx.cache === nothing ? (cx.cache = IdDict()) : cx.cache
 
+"""
+  NoContext <: Zygote.AContext
+
+The `NoContext` context assumes that there is no global context, and errors if any part of a
+function that is being differentiated interacts with either globals or tasks.
+
+The benefit of using such a construct is the elimination of overhead associated with the
+`cache` `IdDict` in a standard `Context`, which can be prohibitively large in certain
+settings.
+"""
+struct NoContext <: Zygote.AContext end
+
+# There is no cache, so there are never any fields.
+cache(cx::NoContext) = (cache_fields=nothing)
```

Comment:
Should this error?
Or return a custom type that has setindex overloaded, to give a helpful exception that says you can't use `NoContext`

Reply:
I'm not at all sure whether the current implementation of this stuff is the most logical way to do things -- it's a hack that appears to work.

For example, I'm not even really sure what the extent of the use of `accum_param` is. It appears to be used in the `_pullback` definitions for
1. `unwrap`.
2. `literal_getindex` for `NTuple`s,
3. `getindex` for `NTuple`s
4. `literal_getfield`

I'm not at all clear on the purpose of `unwrap`, so I'm not really sure what I'm breaking there.